src/main/java/org/tugraz/sysds/runtime/instructions/cp/SqlCPInstruction.java

```diff
@@ -0,0 +1,125 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.tugraz.sysds.runtime.instructions.cp;
+
+import org.tugraz.sysds.common.Types.ValueType;
+import org.tugraz.sysds.runtime.DMLRuntimeException;
+import org.tugraz.sysds.runtime.controlprogram.context.ExecutionContext;
+import org.tugraz.sysds.runtime.data.TensorBlock;
+import org.tugraz.sysds.runtime.instructions.InstructionUtils;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+public class SqlCPInstruction extends CPInstruction {
+	private CPOperand _conn, _user, _pass, _query, _output;
+	
+	public SqlCPInstruction(CPOperand conn, CPOperand user, CPOperand pass, CPOperand query, CPOperand out,
+			String opcode, String instr) {
+		super(CPType.Sql, opcode, instr);
+		_conn = conn;
+		_user = user;
+		_pass = pass;
+		_query = query;
+		_output = out;
+	}
+	
+	public static SqlCPInstruction parseInstruction(String str) {
+		String[] parts = InstructionUtils.getInstructionPartsWithValueType(str);
+		String opcode = parts[0];
+		
+		if( parts.length != 6 )
+			throw new DMLRuntimeException("Invalid number of operands in sql instruction: " + str);
+		
+		CPOperand conn, user, pass, query, out;
+		conn = new CPOperand(parts[1]);
+		user = new CPOperand(parts[2]);
+		pass = new CPOperand(parts[3]);
+		query = new CPOperand(parts[4]);
+		out = new CPOperand(parts[5]);
+		return new SqlCPInstruction(conn, user, pass, query, out, opcode, str);
+	}
+	
+	@Override
+	public void processInstruction(ExecutionContext ec) {
+		String conn = ec.getScalarInput(_conn).getStringValue();
+		String user = ec.getScalarInput(_user).getStringValue();
+		String pass = ec.getScalarInput(_pass).getStringValue();
+		String query = ec.getScalarInput(_query).getStringValue();
+		
+		try (Connection connection = user.isEmpty() ? DriverManager.getConnection(conn) :
+				DriverManager.getConnection(conn, user, pass)) {
+			Statement statement = connection.createStatement();
+			ResultSet resultSet = statement.executeQuery("SELECT COUNT(*) FROM (" + query + ") AS sub");
+			resultSet.next();
+			int rows = resultSet.getInt(1);
+			resultSet = statement.executeQuery(query);
+			ResultSetMetaData meta = resultSet.getMetaData();
+			int cols = meta.getColumnCount();
+			ValueType[] schema = getSchemaFromMetaData(meta);
+			
+			int[] dims = {rows, cols};
+			TensorBlock outBlock = new TensorBlock(schema, dims);
+			for (int row = 0; resultSet.next(); row++)
+				for (int i = 0; i < cols; i++)
+					setCell(outBlock, resultSet, schema[i], new int[]{row, i});
```

Comment:
This cell by cell copy may not work the best for bigger relations. Not sure what I am looking for, but we may need a block by block copy (something along the line of System.arraycopy() ?).